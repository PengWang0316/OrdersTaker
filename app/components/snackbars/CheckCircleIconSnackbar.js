import React from 'react';
import PropTypes from 'prop-types';
import { Snackbar, SnackbarContent } from '@material-ui/core';
import { CheckCircle } from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';

/* istanbul ignore next */
const styles = theme => ({
  messageSpan: {
    display: 'flex',
    alignItems: 'center'
  },
  message: {
    marginLeft: 15
  },
  snackbarContent: {
    backgroundColor: theme.palette.primary.dark
  }
});

export const CheckCircleIconSnackbar = ({
  open, classes, onClose, message
}) => (
  <Snackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    open={open}
    onClose={onClose}
    autoHideDuration={2000}
    ContentProps={{
      'aria-describedby': 'message-id',
    }}
  >
    <SnackbarContent
      className={classes.snackbarContent}
      message={
        <span id="message-id" className={classes.messageSpan}>
          <CheckCircle />
          <span className={classes.message}>{message}</span>
        </span>
      }
    />
  </Snackbar>
);
CheckCircleIconSnackbar.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  message: PropTypes.string.isRequired
};
export default withStyles(styles)(CheckCircleIconSnackbar);
