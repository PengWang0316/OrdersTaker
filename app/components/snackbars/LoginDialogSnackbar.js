import React from 'react';
import PropTypes from 'prop-types';
import { Snackbar, IconButton, SnackbarContent } from '@material-ui/core';
import { Close, Error } from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';

/* istanbul ignore next */
const styles = theme => ({
  messageSpan: {
    display: 'flex',
    alignItems: 'center'
  },
  message: {
    marginLeft: 15
  },
  snackbarContent: {
    backgroundColor: theme.palette.error.dark
  }
});

export const LoginDialogSnackbar = ({
  message, onClose, open, classes
}) => (
  <Snackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    open={open}
    autoHideDuration={4000}
    onClose={onClose}
    ContentProps={{
      'aria-describedby': 'message-id',
    }}
  >
    <SnackbarContent
      className={classes.snackbarContent}
      message={
        <span id="message-id" className={classes.messageSpan}>
          <Error />
          <span className={classes.message}>{message}</span>
        </span>
      }
      action={[
        <IconButton
          key="close"
          aria-label="Close"
          color="inherit"
          onClick={onClose}
        >
          <Close />
        </IconButton>,
      ]}
    />
  </Snackbar>
);
LoginDialogSnackbar.propTypes = {
  message: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  open: PropTypes.bool.isRequired
};
export default withStyles(styles)(LoginDialogSnackbar);
