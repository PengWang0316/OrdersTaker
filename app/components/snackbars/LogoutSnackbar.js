import React from 'react';
import PropTypes from 'prop-types';
import { Snackbar, SnackbarContent } from '@material-ui/core';
import { CheckCircle } from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';

/* istanbul ignore next */
const styles = theme => ({
  messageSpan: {
    display: 'flex',
    alignItems: 'center'
  },
  message: {
    marginLeft: 15
  },
  snackbarContent: {
    backgroundColor: theme.palette.primary.dark
  }
});

export const LogoutSnackbar = ({
  open, classes, onClose
}) => (
  <Snackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    open={open}
    onClose={onClose}
    autoHideDuration={2000}
    ContentProps={{
      'aria-describedby': 'message-id',
    }}
  >
    <SnackbarContent
      className={classes.snackbarContent}
      message={
        <span id="message-id" className={classes.messageSpan}>
          <CheckCircle />
          <span className={classes.message}>Logout Successfully</span>
        </span>
      }
    />
  </Snackbar>
);
LogoutSnackbar.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired
};
export default withStyles(styles)(LogoutSnackbar);
