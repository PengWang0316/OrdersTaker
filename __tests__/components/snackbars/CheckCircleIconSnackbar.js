import React from 'react';
import renderer from 'react-test-renderer';

import { CheckCircleIconSnackbar } from '../../../app/components/snackbars/CheckCircleIconSnackbar';

jest.mock('@material-ui/core/Snackbar', () => 'Snackbar');
jest.mock('@material-ui/core/SnackbarContent', () => 'SnackbarContent');
jest.mock('@material-ui/icons/CheckCircle', () => 'CheckCircle');

describe('CheckCircleIconSnackbar', () => {
  const defaultProps = {
    open: true,
    onClose: jest.fn(),
    classes: {
      messageSpan: 'messageSpan',
      message: 'message',
      snackbarContent: 'snackbarContent'
    },
    message: 'message'
  };
  test('Snapshot', () => expect(renderer.create(<CheckCircleIconSnackbar {...defaultProps} />).toJSON()).toMatchSnapshot());
});
