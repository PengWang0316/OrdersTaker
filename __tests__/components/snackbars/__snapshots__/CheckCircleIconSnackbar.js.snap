// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckCircleIconSnackbar Snapshot 1`] = `
<Snackbar
  ContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "bottom",
    }
  }
  autoHideDuration={2000}
  onClose={[MockFunction]}
  open={true}
>
  <SnackbarContent
    className="snackbarContent"
    message={
      <span
        className="messageSpan"
        id="message-id"
      >
        <CheckCircle />
        <span
          className="message"
        >
          message
        </span>
      </span>
    }
  />
</Snackbar>
`;
